### PR TITLE
Fixed default config for `Pix2Struct` model to set `Pix2StructTextModel` to `is_decoder=True`

### PR DESCRIPTION
**Pull request by gbarello-uipath**
Previously, the `Pix2StructTextModel` was configured with `is_decoder=False` by default causing the attention mask used for self-attention to be non-causal and causing fine-tuning to fail.

As a fix, this PR adds `is_decoder=True` default kwarg to the `Pix2StructTextConfig` class in order to correctly configure the text model as a decoder. 

Fixes # 22903

@younesbelkada

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this! I can confirm this has no impact on the current slow tests
I didn't checked the doctests yet but I can take care of that in a follow up PR, I am also updating all model checkpoints. We can merge that meanwhile
cc @sgugger

**Review by sgugger — APPROVED** [review]

**Comment by RainbowMan1** [comment]
Shouldn't the matcha and deplot checkpoints be updated as well?

**Comment by younesbelkada** [comment]
Good point, will update them now

**Comment by younesbelkada** [comment]
Just updated them! Thanks for flagging @RainbowMan1
